packages/store/addon/-private/system/snapshot.js

```diff
@@ -85,6 +90,29 @@ export default class Snapshot {
     return this._internalModel.getRecord();
   }
 
+  get _attributes() {
+    let attributes = this.__attributes;
+
+    if (attributes === null) {
+      let record = this.record;
+      attributes = this.__attributes = Object.create(null);
+      let attrs = Object.keys(this._store._attributesDefinitionFor(this.modelName, record.id));
+      if (CUSTOM_MODEL_CLASS) {
+        attrs.forEach(keyName => {
+          if (this.type.isModel) {
+            attributes[keyName] = get(record, keyName);
+          } else {
+            attributes[keyName] = this._internalModel._recordData.getAttr(keyName);
```

Comment:
lets use recordData directly instead of relying on _internalModel, either via stashing a prop or helper lookup